File: components/org.apache.stratos.messaging/src/main/java/org/apache/stratos/messaging/event/tenant/SubscriptionDomainAddedEvent.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.apache.stratos.messaging.event.tenant;

import org.apache.stratos.messaging.event.Event;

import java.io.Serializable;
import java.util.*;

/**
 * This event is fired when domains are added to a tenant subscription.
 */
public class SubscriptionDomainAddedEvent extends Event implements Serializable {
    private static final long serialVersionUID = 3457484382856403382L;

    private final int tenantId;
    private final String serviceName;
    private final Set<String> clusterIds;
    private final String domainName;
    private final String applicationContext;

    public SubscriptionDomainAddedEvent(int tenantId, String serviceName, Set<String> clusterIds, String domainName,
                                        String applicationContext) {
        this.tenantId = tenantId;
        this.serviceName = serviceName;
        this.clusterIds = clusterIds;
        this.domainName = domainName;
        this.applicationContext = applicationContext;
    }

    public int getTenantId() {
        return tenantId;
    }

    public String getServiceName() {
        return serviceName;
    }

    public Set<String> getClusterIds() {
        return Collections.unmodifiableSet(clusterIds);
    }

    public String getDomainName() {
        return domainName;
    }

    public String getApplicationContext() {
        return applicationContext;
    }
}

File: tools/python-cartridge-agent/cartridge-agent/event/tenant/subscriptiondomainaddedevent.py
import json


class SubscriptionDomainAddedEvent():

    def __init__(self):
        self.tenant_id = None
        self.service_name = None
        self.cluster_ids = None
        self.domain_name = None
        self.application_context = None

    @staticmethod
    def create_from_json(json_str):
        json_obj = json.loads(json_str)
        instance = SubscriptionDomainAddedEvent()

        instance.cluster_ids = json_obj["clusterIds"] if "clusterIds" in json_obj else None
        instance.tenant_id = json_obj["tenantId"] if "tenantId" in json_obj else None
        instance.service_name = json_obj["serviceName"] if "serviceName" in json_obj else None
        instance.domain_name = json_obj["domainName"] if "domainName" in json_obj else None
        instance.application_context = json_obj["applicationContext"] if "applicationContext" in json_obj else None

        return instance